examples/C++/DDS/ContentFilterTopic/CustomFilterSubscriber.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomFilterSubscriber.cpp
+ *
+ */
+
+#include "CustomFilterSubscriber.hpp"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/SubscriberAttributes.h>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+
+using namespace eprosima::fastdds::dds;
+
+bool CustomFilterSubscriber::init()
+{
+    DomainParticipantQos pqos;
+    pqos.name("Participant_sub");
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(0, pqos);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    // Register the filter factory
+    if (ReturnCode_t::RETCODE_OK !=
+            participant_->register_content_filter_factory("MY_CUSTOM_FILTER", &filter_factory))
+    {
+        // Error
+        return false;
+    }
+
+    //Register the type
+    type_.register_type(participant_);
+
+    //CREATE THE SUBSCRIBER
+    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+
+    if (subscriber_ == nullptr)
+    {
+        return false;
+    }
+
+    //Create the topic
+    topic_ = participant_->create_topic(
+        "HelloWorldTopic",
+        "HelloWorld",
+        TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // Create a ContentFilteredTopic using an expression with no parameters
+    std::string expression = " ";
+    std::vector<std::string> parameters;
+    parameters.push_back("3");
+    parameters.push_back("5");
```

Comment:
I suggest using another parameters so this filter has a different behavior from the DefaultSQLFilter